packages/npm-resolver/src/pickPackageFromMeta.ts

```diff
@@ -76,6 +77,17 @@ function pickVersionByVersionRange (
         }
         break
       }
+      // This is a special type of "preference". `versionPref` is the version
+      // range written by yarn into the lockfile. `originalPref` is what pnpm
+      // found in the requesting package.json. If they match, we should exit
+      // early and use the version resolved in the lockfile.
+      case 'lockfile': {
+        const [versionPref, resolvedVersion] = preferredSelector.split('@')
```

Comment:
why do you need to parse it? Why don't you just save it in an object instead of a string?

Reply:
I was worried about dealing with the blast radius of changing the [`VersionSelectors`](https://github.com/pnpm/pnpm/blob/5581e4ef62152c133c6222b780c915eb12c3a4c6/packages/resolver-base/src/index.ts#L48) type signature. The selector itself if the key in the object which makes it challenging to modify.